### ISSUE TITLE
Feature Request: Masking the program in task manager

### ISSUE THREAD

**Issue by RushilKoul** [xp4xbox/Python-Backdoor#71]
A really cool feature would be having an ability to hide the program in the task manager, or change the process name and/or icon to something else. Right now, A user can easily disable the backdoor by simply using the task manager.

**Comment by xp4xbox** [comment]
There is no way to hide a process from task manager, but there might be something else. Il look into it later.

**Comment by Hawkern** [comment]
Or is there?

**Comment by xp4xbox** [comment]
Not possible.

**Comment by hemish04082005** [comment]
It is possible to give a custom icon using pyinstaller and giving a fake name which really looks like an initial componnet like Google service handler, explorer.exe or something like that

**Comment by xp4xbox** [comment]
Yeah, in the setup use the melt file option, and then pick an icon.

**Comment by gratisCobalt** [comment]
only thing I thought of is registering it as a service which will hide it from taskmanager task tab (would only be visible in taskmanager services tab) and it wont show on startup tab

**Comment by xp4xbox** [comment]
I will look into it.

**Comment by Hasham14** [comment]
Can we _migrate_ the process like in meterpreter

**Comment by xp4xbox** [comment]
> Can we _migrate_ the process like in meterpreter

Forgot to respond to this, but no it is not possible, since migrating the process works by basically writing the payload into the process (would need to be able to have this work as shellcode)

**Comment by xp4xbox** [comment]
> only thing I thought of is registering it as a service which will hide it from taskmanager task tab (would only be visible in taskmanager services tab) and it wont show on startup tab

I looked into this earlier, it is not possible without admin at least.

**Comment by xp4xbox** [comment]
Leaving it open incase there is another idea.

**Comment by gratisCobalt** [comment]
just an idea... you could inject it into an unsuspicious process like e.g. notepad (in taskmanager it should show the process as e.g. notepad.exe) but maybe it will lead into getting detected easier
